File: Programmers/Level6/eatCookie(cjpt)/EatCookie.java
class EatCookie
{
	public int eatCookie(int []cookies){
		int answer = 0;
		int[] dp = new int[cookies.length];
		for(int i=0; i<dp.length; ++i)
		{
			dp[i] = 1;
		}

		for(int i=1; i<cookies.length; ++i)
		{
			for(int j=0; j<i; ++j)
			{
				if(cookies[i] > cookies[j])
				{
					if(dp[i] < dp[j]+1)
					{
						dp[i] = dp[j]+1;
					}
				}
			}
		}

		for(int i=0; i<dp.length; ++i)
		{
			answer = Math.max(answer,dp[i]);
		}

		return answer;
	}

	public static void main(String[] args) {
		EatCookie e = new EatCookie();
		int []cookies = {1, 4, 2, 6, 3, 4, 1, 5};
		System.out.println(e.eatCookie(cookies));
	}
}

File: Programmers/Level6/eatCookie(cjpt)/eatCookie.py
def eatCookie(cookies):
    answer = 0
    now = []
    def helper(cookies):
        nonlocal answer        
        for i, e in enumerate(cookies):
            if not bool(now) or now[-1] < e:
                now.append(e)
                helper(cookies[i+1:])
                now.pop()
        answer = max(answer,len(now))
        return
    helper(cookies)
    return answer



# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(eatCookie([1, 4, 2, 6, 3, 4, 1, 5]))